### DISCUSSION TITLE
AGH connects to server very far from my location

### DISCUSSION THREAD

**Discussion by PaperCutBad** [AdguardTeam/AdGuardHome#3393] [Q&A] [answered]
I am using AGH on Windows 10 installed as service. I use NextDNS HTTPS but it connects to France nextdns server even though there is nextdns server in Delhi which is 200KM from my location. This results in very high RTT but AdGuard Windows connects to Delhi server. If anyone have faced this issue, is there any solution to it?

**Comment by emlimap** [comment, marked as answer]
I have this issue as well. The way NextDNS steers traffic seems to be a bit erratic. One day it will be fine and connect to the closest server but another day a server in another country, still close to my home country so isn't terrible. In the worst case (only happened a few times), it would connect to a server on the other side of the world.

The way I worked around it is to run their CLI client(https://github.com/nextdns/nextdns) on another port and get Adguard Home to use it as upstream DNS. Their client connects to the server with the lowest latency which you can see in its output.

**Reply by PaperCutBad** [reply]
For me it always connects to server on other side of world. I tried NextDNS CLI client too but it just disconnects me from internet every now and then. Currently I am using Technitium instead of AdGuard which connects to the closest server in my country and have almost all features of AGH but GUI isn't that good and query log is available only as txt file.

**Reply by emlimap** [reply]
You could try setting the bootstrap DNS server to a Public DNS server that supports [ECS](https://en.wikipedia.org/wiki/EDNS_Client_Subnet) such as Google DNS or OpenDNS. You can even try setting your ISP DNS servers as bootstrap DNS server as another test.

Failing that you will have to contact NextDNS as per instructions from their diagnostics tool https://help.nextdns.io/t/y4hmvcx/report-network-latency-issue

**Reply by PaperCutBad** [reply]
> You could try setting the bootstrap DNS server to a Public DNS server that supports [ECS](https://en.wikipedia.org/wiki/EDNS_Client_Subnet) such as Google DNS or OpenDNS. You can even try setting your ISP DNS servers as bootstrap DNS server as another test.
> 

Thanks this worked. I tried using NextDNS IPv4 address as bootstrap DNS server and now it connects to nearest server.
